discovery-provider/alembic/versions/f82eb376f471_add_safe_to_date_func.py

```diff
@@ -0,0 +1,52 @@
+"""add_safe_to_date_func
+
+Creates a func `to_date_safe` that behaves like `to_date` except
+NULL and empty-string input value are coerced to a NULL response rather
+than throwing an exception.
+
+Revision ID: f82eb376f471
+Revises: 6d1b38f242fe
+Create Date: 2020-12-23 18:54:29.965280
+
+"""
+from alembic import op
+
+
+# revision identifiers, used by Alembic.
+revision = 'f82eb376f471'
+down_revision = '6d1b38f242fe'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    connection = op.get_bind()
+    connection.execute(
+        """
+        CREATE OR REPLACE FUNCTION to_date_safe(p_date VARCHAR, p_format VARCHAR)
+        RETURNS DATE
+        LANGUAGE plpgsql
+        as $$
+        DECLARE
+            ret_date DATE;
+        BEGIN
+            IF p_date = '' THEN
```

Comment:
strictly speaking you dont need a migration if you just want to do this if check. postgres has a function to coerce empty string to null with NULLIF like 
```COALESCE(TO_DATE(NULLIF("release_date", ''), 'Dy Mon DD YYYY HH24:MI:SS'), "created_at")```

Reply:
try this sql for example 
```
SELECT "release_date", COALESCE(TO_DATE("release_date", 'Dy Mon DD YYYY HH24:MI:SS'), "created_at") as "release_date_coalesce", COALESCE(TO_DATE(NULLIF("release_date", ''), 'Dy Mon DD YYYY HH24:MI:SS'), "created_at") as "release_date_coerce_coalesce" FROM "tracks"  WHERE "release_date" = '' LIMIT 10;
```

Reply:
Yeah, I wish I could but the problem is if there's garbage in there which is possible since it's a string field and exception gets thrown and the whole query fails. Can you think of a way to get around that? (See the test case with the garbage string)

Reply:
I think the real solution here is to do validation on the field before writing to db and migrating this field to a date time but that's probably beyond the scope of this bug fix for now